### DISCUSSION TITLE
run tokio-1.12.0 test failed!

### DISCUSSION THREAD

**Discussion by RustChater** [tokio-rs/tokio#4144] [Q&A]
When I run a test:

cd tokio-1.12.1
cargo test --color=always --package tokio --test tcp_echo echo_server
error: There are multiple `tokio` packages in your project, and the specification `tokio` is ambiguous.
Please re-run this command with `-p <spec>` where `<spec>` is one of the following:
  file:///Users/xxx/tmp/tokio-1.12.0#tokio:1.12.0
  https://github.com/rust-lang/crates.io-index#tokio:1.12.0
Process finished with exit code 101

Follow the prompts, added argument -p, but still got same error.
cargo test --color=always --package tokio --test tcp_echo echo_server -p "file:///Users/xxx/tmp/tokio-1.12.0#tokio:1.12.0"

env:
macos 11.6
stable-x86_64-apple-darwin (default)
rustc 1.55.0 (c8dfcfe04 2021-09-06)

**Comment by Darksonn** [comment]
I don't really understand what is happening here. There's no `tokio-1.12.1` folder in Tokio (there isn't even a version by that name). 

The way you run tests is to clone the repo, enter the `tokio` repo and run
```
cargo test --all-features
```

**Reply by RustChater** [reply]
It's no problem if git clone master.
thanks your answer!
